docs(contributing): quick start (#31333)
aws/aws-cdk: CONTRIBUTING.md

1. [Publib](https://github.com/cdklabs/publib) - A toolchain for publishing packages to multiple repositories. A lot of this is included in projen and we recommend using that instead of publib directly, but it may be useful for specific cases.
1. [Construct Hub](https://constructs.dev) - An index of all construct libraries published to NPM. When you publish a construct library, it will automatically have documentation generated and published to Construct Hub.

## Quick Start

> Fork the aws-cdk repository into your account: https://github.com/aws/aws-cdk/fork

Reviewer: ```suggestion
### Setup
Fork the aws-cdk repository into your account: https://github.com/aws/aws-cdk/fork
```

Clone the forked repository:
```console
$ git clone https://github.com/{your-account}/aws-cdk.git
$ cd aws-cdk
$ yarn install
```

Before you create a pull request:
* Write code changes
* Write unit tests
* Write integ tests
* Commit changes and push to remote branch

Build the entire aws-cdk repo (this may take some time):
```console

> $ npx lerna run build --skip-nx-cache

Reviewer: ```suggestion
$ npx lerna run build
```

The details of flags can be discussed below

```

> Run the unit tests for the modules(e.g. aws-lambda) you've changed:

Reviewer: ```suggestion
### Testing
Run the unit tests for the modules(e.g. aws-lambda) you've changed:
```

```console
$ cd aws-cdk/packages/aws-cdk-lib
$ yarn test aws-lambda
```

Run the integration tests for the modules(e.g. aws-lambda) you've changed:
```console
$ cd aws-cdk/packages/@aws-cdk-testing/framework-integ
$ yarn integ test/aws-lambda/test/integ.lambda.js --update-on-failed
```

> If there are sample code changes in README.md file:

Reviewer: ```suggestion
If you've made changes to sample code in any README, ensure those examples compile with:
```

```console
$ /bin/bash ./scripts/run-rosetta.sh
```

> Link your CDK application to the forked AWS-CDK repository:

Reviewer: ```suggestion
### Linking
If you would like to test your code changes against a CDK App, create the App and link your local CDK with it:
```

```console
$ mkdir cdkApp # in parent dir of aws-cdk
$ cd cdkApp
$ npx cdk@latest init app --language typescript

> $ npx cdk --version # show current version e.g. 2.155.0 (build 34dcc5a)

Reviewer: ```suggestion
$ npx cdk --version # shows the latest CDK version e.g. 2.155.0 (build 34dcc5a)
```
